File: Excel/649 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/649 Conditional Running Total.xlsx"
input = read_excel(path, range = "A1:C20")
test  = read_excel(path, range = "D1:D20")

result = input %>%
    mutate(or_ind = row_number()) %>%
    arrange(Group) %>%
    mutate(Value = ifelse(Reset == "Y", 0, Value)) %>%
    mutate(inner_group = cumsum(Reset == "Y")) %>% 
    group_by(Group, inner_group) %>%
    mutate(`Answer Expected` = cumsum(Value)) %>%
    ungroup() %>%
    arrange(or_ind)

all.equal(result$`Answer Expected`, test$`Answer Expected`)
# [1] TRUE

File: Excel/600-699/649/649 Challenge.py
import pandas as pd

path = "649 Conditional Running Total.xlsx"
input = pd.read_excel(path, usecols="A:C", nrows=20)
test = pd.read_excel(path, usecols="D:D", nrows=20)

input['or_ind'] = input.index + 1
input = input.sort_values(by=['Group', 'or_ind'])
input['Value'] = input.apply(lambda row: 0 if row['Reset'] == 'Y' else row['Value'], axis=1)
input['inner_group'] = input['Reset'].eq('Y').cumsum()
input['Answer Expected'] = input.groupby(['Group', 'inner_group'])['Value'].cumsum()
input = input.sort_values(by='or_ind').reset_index(drop=True)

print(input['Answer Expected'].equals(test['Answer Expected'])) # True